pkg/streamx/provider/ttstream/client_provier.go

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2024 CloudWeGo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ttstream
+
+import (
+	"context"
+	"runtime"
+
+	"github.com/bytedance/gopkg/cloud/metainfo"
+	"github.com/cloudwego/gopkg/protocol/ttheader"
+
+	"github.com/cloudwego/kitex/pkg/kerrors"
+	"github.com/cloudwego/kitex/pkg/rpcinfo"
+	"github.com/cloudwego/kitex/pkg/serviceinfo"
+	"github.com/cloudwego/kitex/pkg/streamx"
+	"github.com/cloudwego/kitex/pkg/streamx/provider/ttstream/ktx"
+)
+
+var _ streamx.ClientProvider = (*clientProvider)(nil)
+
+// NewClientProvider return a client provider
+func NewClientProvider(sinfo *serviceinfo.ServiceInfo, opts ...ClientProviderOption) (streamx.ClientProvider, error) {
```

Comment:
Actually I don't really want users to specify serviceInfo by themselves, because serviceInfo is a pointer and we should not expose it to users explicitly (right?)
Do you think there's a way to prevent users from specifying it?

Reply:
@Marina-Sakai our current code already expose serviceinfo in gen code. User shuld use the gen code's serviceinfo.

Reply:
@YangruiEmma I remember you said you didn't want users to use serviceinfo by themselves even if we expose serviceinfo to users before, didn't you?